frontend/components/pricing/PremiumProductBox.vue

```diff
@@ -117,82 +113,84 @@ const planButton = computed(() => {
   let isDowngrade = false
   let text = $t('pricing.premium_product.button.select_plan')
 
-  if (isLoggedIn.value) {
-    if (currentPremiumSubscription.value) {
-      const subscribedProduct = products.value?.premium_products.find(
-        product =>
-          product.product_id_monthly
-          === currentPremiumSubscription.value!.product_id
-          || product.product_id_yearly
-          === currentPremiumSubscription.value!.product_id,
-      )
-      if (
-        currentPremiumSubscription.value.product_id
-        === props.product.product_id_monthly
-        || currentPremiumSubscription.value.product_id
-        === props.product.product_id_yearly
-        || subscribedProduct === undefined
-      ) {
-        // (this box is either for the subscribed product)
-        // || (the user has an unknown product, possible from V1 or maybe a custom plan)
-        text = $t('pricing.premium_product.button.manage_plan')
-      }
-      else if (
-        subscribedProduct.price_per_month_eur
-        < props.product.price_per_month_eur
-      ) {
-        text = $t('pricing.premium_product.button.upgrade')
-      }
-      else {
-        isDowngrade = true
-        text = $t('pricing.premium_product.button.downgrade')
-      }
+  if (currentPremiumSubscription.value) {
+    const subscribedProduct = products.value?.premium_products.find(
+      product =>
+        product.product_id_monthly
+        === currentPremiumSubscription.value!.product_id
+        || product.product_id_yearly
+        === currentPremiumSubscription.value!.product_id,
```

Comment:
issue: formatting is hard to read and it works without disabling typescript `!`
```suggestion
      product =>
        product.product_id_monthly === currentPremiumSubscription.value?.product_id
        || product.product_id_yearly === currentPremiumSubscription.value?.product_id,
```

Reply:
Done